src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -177,6 +177,110 @@ internal static unsafe HttpMethod GetKnownMethod(byte* data, int length, out int
             return HttpMethod.Custom;
         }
 
+        /// <summary>
+        /// Parses string <paramref name="value"/> for a known HTTP method.
+        /// </summary>
+        /// <remarks>
+        /// A "known HTTP method" can be an HTTP method name defined in the HTTP/1.1 RFC.
+        /// The Known Methods (CONNECT, DELETE, GET, HEAD, PATCH, POST, PUT, OPTIONS, TRACE)
+        /// </remarks>
+        /// <returns><c>true</c> if the input matches a known string, <c>false</c> otherwise.</returns>
+        public static bool GetKnownMethod(string value, out HttpMethod method)
+        {
+            // Only called if user code overrides and sets the Method on IHttpRequestFeature
+            // This should be a rare occurance so not performance sensitive also needs to validate inputs
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            var length = value.Length;
+            if (length == 0)
+            {
+                throw new ArgumentException(nameof(value));
+            }
+
+            var firstChar = value[0];
```

Comment:
Why this implementation if its no performance sensitive? Just use the http abstractions methods

Reply:
Is used by http2 so may be perf sensitive :-/

Also are aspnet dlls ngen'd in the newest versions? Does mean none of the methods can be inlined if so.

Did you mean
```csharp
if (HttpMethods.IsGet(value))
{
    method = HttpMethod.Get;
    return true;
}
```

Reply:
How is this used by HTTP/2? If this is only hit by applications setting the request method themselves, I too would prefer a more readable slightly slower version of this method.

Reply:
HTTP/2 goes via string header
```csharp
((IHttpRequestFeature)this).Method = RequestHeaders[":method"];
```
Though it probably shouldn't use an interface cast for it https://github.com/dotnet/coreclr/issues/16198

Reply:
e.g. first line of `Http2Stream.TryParseRequest(ReadResult result, out bool endConnection)`

Reply:
Improved the `TryParseRequest` not to cast via interface and use method directly

Reply:
Also changed to use the `HttpMethods` strings